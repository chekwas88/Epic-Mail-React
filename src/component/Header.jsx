import React from 'react';


/**
 * @method Header
 * @param {object} event React synthetic event object
 * @description Header component
 * @returns {JSX} JSX Markup
 */
const Header = () => (
  <header>
    <nav className="navbar">
      <div className="burger">
        <div>
          <span />
          <span />
          <span />
        </div>

      </div>
      <div className="logo" />
      <ul className="header-links">
        <li>
            login
        </li>
      </ul>
    </nav>
  </header>
);
export default Header;
