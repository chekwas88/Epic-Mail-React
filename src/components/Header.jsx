/* eslint-disable react/prefer-stateless-function */
import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import { bool } from 'prop-types';


/**
 * @method Header
 * @param {object} event React synthetic event object
 * @description Header component
 * @returns {JSX} JSX Markup
 */
export class Header extends Component {
  /**
   * @method render
   * @description React render method
   * @returns {JSX} React component markup
   */
  render() {
    const { isLoggedIn } = this.props;
    let headerContent = '';
    if (isLoggedIn) {
      headerContent = (
        <Fragment>
          <li>
            <Link to="/inbox">Inbox</Link>
          </li>
          <li>
            <Link to="/sent">Sent</Link>
          </li>
          <li>
            <Link to="/groups">Groups</Link>
          </li>
        </Fragment>
      );
    }
    return (
      <header>
        <nav className="navbar">
          <div className="logo" />
          <ul className="header-links">
            {headerContent}
          </ul>
        </nav>
      </header>
    );
  }
}

/**
 * @method mapStateToProps
 * @description maps reducer states to props
 * @param {object} * destructured reducer state object
 * @returns {object} state
 */
export const mapStateToProps = ({ auth }) => {
  const { isLoggedIn } = auth;
  return {
    isLoggedIn,
  };
};
export default connect(mapStateToProps)(Header);

Header.propTypes = {
  isLoggedIn: bool.isRequired,
};
