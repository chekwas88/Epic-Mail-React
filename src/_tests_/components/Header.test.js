/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import { shallow } from 'enzyme';
import Header from '../../components/Header';

describe('Test for Header component', () => {
  test('should render header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
  });
});
