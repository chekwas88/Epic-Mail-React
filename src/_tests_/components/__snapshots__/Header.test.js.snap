// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header component should render header correctly 1`] = `
<header>
  <nav
    className="navbar"
  >
    <div
      className="burger"
    >
      <div>
        <span />
        <span />
        <span />
      </div>
    </div>
    <div
      className="logo"
    />
    <ul
      className="header-links"
    >
      <li>
        login
      </li>
    </ul>
  </nav>
</header>
`;
